### PR TITLE
Autreg 99 upload data

### PR DESCRIPTION
**Pull request by Eiriknil**

### CODE REVIEW

File: api/src/main/java/no/bibsys/web/DatabaseResource.java [hunk 1]
```diff
@@ -341,6 +348,45 @@ public Response entitiesSummary(@HeaderParam(ApiKeyConstants.API_KEY_PARAM_NAME)
...
+                    schema = @Schema(type = STRING)) @PathParam(REGISTRY_NAME) String registryName,
+            @RequestBody(description = "Array of Entity to upload",
+                    content = @Content(array = @ArraySchema(schema = @Schema(implementation = EntityDto.class)))) EntityDto... entityDtos)
+            throws IOException {
+
+        
+        
+        List<EntityDto> persistedEntities = new ArrayList<EntityDto>();
+        Arrays.asList(entityDtos).forEach(entityDto -> {
+            EntityDto persistedEntity = entityService.addEntity(registryName, entityDto);
```
Comment (axthosarouris) [inline api/src/main/java/no/bibsys/web/DatabaseResource.java]:
```
            List<EntityDto> persistedEntities = Arrays.stream(entityDtos)
            .map(entityDto -> persistEntity(registryName, entityDto))
            .collect(Collectors.toList());
```
where `persitEntity`      
```
    private EntityDto persistEntity(String registryName,EntityDto entityDto) {
        EntityDto persistedEntity = entityService.addEntity(registryName, entityDto);
        String entityId = persistedEntity.getId();
        persistedEntity.setPath(String.join("/", "registry", registryName, "entity", entityId));
        return persistedEntity;
    }
```

File: api/src/main/java/no/bibsys/web/DatabaseResource.java [hunk 2]
```diff
@@ -361,6 +362,7 @@ public Response entitiesSummary(@HeaderParam(ApiKeyConstants.API_KEY_PARAM_NAME)
                     value = AwsApiGatewayIntegration.AWS_PROXY), }) })
     @SecurityRequirement(name = ApiKeyConstants.API_KEY)
     @RolesAllowed({ Roles.API_ADMIN, Roles.REGISTRY_ADMIN })
+    @Produces("application/json")
```
Comment (brinxmat) [inline api/src/main/java/no/bibsys/web/DatabaseResource.java]: Can we use something like javax.ws.rs.core.MediaType.APPLICATION_JSON? Possibly own class/enum if applicaiton/ld+json is to be supported.

File: api/src/main/java/no/bibsys/web/DatabaseResource.java [hunk 3]
```diff
@@ -369,15 +371,16 @@ public Response uploadEntities(@HeaderParam(ApiKeyConstants.API_KEY_PARAM_NAME)
...
             EntityDto persistedEntity = entityService.addEntity(registryName, entityDto);
             String entityId = persistedEntity.getId();
             persistedEntity.setPath(String.join("/", "registry", registryName, "entity", entityId));
+            persistedEntities.add(persistedEntity);
         });
 
-
-        return Response.ok(persistedEntities).build();
+        
+        return Response.status(Status.OK).entity(new GenericEntity<List<EntityDto>>(persistedEntities) {}).type(MediaType.APPLICATION_JSON).build();
```
Comment (brinxmat) [inline api/src/main/java/no/bibsys/web/DatabaseResource.java]: C.f comment above

File: api/src/test/java/no/bibsys/web/DatabaseResourceTest.java [hunk 4]
```diff
@@ -303,6 +311,51 @@ public void updateEntity_EntityExists_ReturnsUpdatedEntity() throws Exception {
...
+    public void uploadArrayOfThreeEntities_RegistryExists_RegistryContainsThreeEntities() throws Exception {
+        
+        String registryName = UUID.randomUUID().toString();
+        RegistryDto registryDto = sampleData.sampleRegistryDto(registryName);
+        createRegistry(registryDto);
+        
+        
+        List<EntityDto> sampleEntities = new CopyOnWriteArrayList<EntityDto>();
+        EntityDto sampleEntityDto1 = sampleData.sampleEntityDto();
+        sampleEntityDto1.setId("id1");
```
Comment (brinxmat) [inline api/src/test/java/no/bibsys/web/DatabaseResourceTest.java]: Never keen on magic strings and numbers — I'd maybe even randomise test data to create a generic test of our expectations.

File: api/src/test/java/no/bibsys/web/DatabaseResourceTest.java [hunk 5]
```diff
@@ -303,6 +311,51 @@ public void updateEntity_EntityExists_ReturnsUpdatedEntity() throws Exception {
...
+        
+        Response response = uploadEntities(registryName, sampleEntities);
+        List<EntityDto> readEntityList = response.readEntity(new GenericType<List<EntityDto>>() {});
+        AtomicInteger numberOfEntities = new AtomicInteger(0);
+        
+        readEntityList.forEach(entity -> {
+            try {
+                readEntity(registryName, entity.getId());
+                numberOfEntities.set(numberOfEntities.incrementAndGet());
+                System.out.println("number: " + numberOfEntities.get());
```
Comment (brinxmat) [inline api/src/test/java/no/bibsys/web/DatabaseResourceTest.java]: Was System.out supposed to be here?

Comment (Eiriknil) [reply api/src/test/java/no/bibsys/web/DatabaseResourceTest.java]: System.out removed

File: api/src/main/java/no/bibsys/web/DatabaseResource.java (hunk 2 above)
Comment (Eiriknil) [reply api/src/main/java/no/bibsys/web/DatabaseResource.java]: yes

File: api/src/test/java/no/bibsys/web/DatabaseResourceTest.java (hunk 4 above)
Comment (Eiriknil) [reply api/src/test/java/no/bibsys/web/DatabaseResourceTest.java]: Randomized the identifiers

**Pull request merged**